version: '3.9'

services:

  traefik:
    image: traefik:v2.11
    container_name: traefik
    ports:
      - "80:80"
      - "443:443"
      - "8082:8082"  # Traefik dashboard
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /{{ traefik_dir }}/traefik/config/traefik.yml:/etc/traefik/traefik.yml:ro
      - /{{ traefik_dir }}/traefik/config/dynamic/dynamic.yml:/etc/traefik/dynamic.yml:ro
      - /{{ base_dir }}/certs:/certs:ro
      - /{{ traefik_dir }}/traefik/logs:/logs
    networks:
      - global_net
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.dashboard.rule=Host(`traefik.tst.local`)"
      - "traefik.http.routers.dashboard.service=api@internal"
      - "traefik.http.routers.dashboard.entrypoints=websecure"
      - "traefik.http.routers.dashboard.tls=true"
      - "traefik.http.routers.dashboard.middlewares=auth@file"
      # - "traefik.http.routers.dashboard.tls.certresolver=dummy"
      - "traefik.http.services.dummy.loadbalancer.server.port=9999"

networks:
  global_net:
    external: true

